app/forms/EnhancedMapping.scala

```diff
@@ -32,4 +34,52 @@ object EnhancedMapping {
       }
     override def unbind(key: String, value: String): Map[String, String] = Map(key -> value)
   }
+
+  object MucrOptionsFormatters {
+
+    lazy val newMucrFormatter: Formatter[String] = new Formatter[String] {
+
+      override def bind(key: String, data: Map[String, String]): Either[Seq[FormError], String] = data.get("createOrAdd") match {
+        case Some(Create) =>
+          data.get("newMucr") match {
```

Comment:
if you use `key` here you won't have to provide two formatters just only one and use 
```
"newMucr" -> Forms.of(mucrFormatter)
"existingMucr" -> Forms.of(mucrFormatter)
``` 
in mapping